crates/accelerate/src/basis/basis_translator/basis_search.rs

```diff
@@ -0,0 +1,228 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use std::cell::RefCell;
+
+use hashbrown::{HashMap, HashSet};
+use pyo3::prelude::*;
+
+use crate::equivalence::{EdgeData, Equivalence, EquivalenceLibrary, Key, NodeData};
+use qiskit_circuit::operations::Operation;
+use rustworkx_core::petgraph::stable_graph::{EdgeReference, NodeIndex, StableDiGraph};
+use rustworkx_core::petgraph::visit::Control;
+use rustworkx_core::traversal::{dijkstra_search, DijkstraEvent};
+
+use super::compose_transforms::{BasisTransformIn, GateIdentifier};
+
+#[pyfunction]
+#[pyo3(name = "basis_search")]
+/// Search for a set of transformations from source_basis to target_basis.
+/// Args:
+///     equiv_lib (EquivalenceLibrary): Source of valid translations
+///     source_basis (Set[Tuple[gate_name: str, gate_num_qubits: int]]): Starting basis.
+///     target_basis (Set[gate_name: str]): Target basis.
+///
+/// Returns:
+///     Optional[List[Tuple[gate, equiv_params, equiv_circuit]]]: List of (gate,
+///         equiv_params, equiv_circuit) tuples tuples which, if applied in order
+///         will map from source_basis to target_basis. Returns None if no path
+///         was found.
+pub(crate) fn py_basis_search(
+    py: Python,
+    equiv_lib: &mut EquivalenceLibrary,
+    source_basis: HashSet<GateIdentifier>,
+    target_basis: HashSet<String>,
```

Comment:
Can't you just do `HashSet<&str>` here? I thought there was `FromPyObject` for &str.